File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
# 최대 공약수
def GCD(a, b):
    if b == 0:
        return a
    else:
        return GCD(b, a % b)


# 최소 공배수
def LCM(a, b, gcd):
    return a * b // gcd


a, b = map(int, input().split())
gcd = GCD(a, b)
print(gcd)
print(LCM(a, b, gcd))

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int gcd = 0; // 최대공약수
        int lcm = 0; // 최소공배수
        
        gcd = solveGCD(n, m);
        lcm = solveLCM(gcd, n * m);
        
        int[] answer = {gcd, lcm};
        return answer;
    }
    
    private int solveGCD(int a, int b){
        // 나머지가 0이면 return
        if (a % b == 0) return b;
        return solveGCD(b, a % b);
    }
    
    // 최소공배수 = 두수의 곱 / 최대공약수
    private int solveLCM(int gcd, int nm){
        return nm / gcd;
    }
}